src/clr-angular/forms/datepicker/date-input.ts

```diff
@@ -186,16 +168,72 @@ export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implement
         return (isPlatformBrowser(this.platformId) && this._datepickerEnabledService.isEnabled) ? "text" : "date";
     }
 
+    //
+    // Output Management
+    // Note: For now we will not emit both clrDateChange and ngControl outputs
+    // at the same time. This requires us to listen to keydown and blur events to figure out
+    // exactly when the Output should be emitted.
+    // Our recommendation right now is to either use clrDate or use ngModel/FormControl.
+    // Do not use both of them together.
+    //
+
+    @Output("clrDateChange") _dateUpdated: EventEmitter<Date> = new EventEmitter<Date>(false);
+
     /**
      * Fires this method when the user changes the input focuses out of the input field.
      */
     @HostListener("change", ["$event.target"])
     onValueChange(target: HTMLInputElement) {
         const value: string = target.value;
-        if (value) {
-            this.inputDate = value.trim();
+        const date: Date = this._dateIOService.isValidInput(value);
+        if (date) {
+            const dayModel: DayModel = new DayModel(date.getFullYear(), date.getMonth(), date.getDate());
+            if (!dayModel.isEqual(this._dateNavigationService.selectedDay)) {
+                this._dateNavigationService.selectedDay = dayModel;
+                this._dateUpdated.emit(date);
```

Comment:
Isn't this whole `if` just `this._dateNavigationService.notifySelectedDayChanged(dayModel)`? It already does the equality test for you too.

Reply:
I am not sure if I understand your question correctly @youdz. Are you saying that we call `notifySelectedDayChanged` from here?

This if is in `onValueChange` from where we just want to emit the Javascript Date Object using the `clrDateChange` Output. We never want to change the what the user is entering here. We let the user enter what he/she wants but emit the Javascript Date object when its valid.

The `notifySelectedDayChanged` subscription on the other hand converts the DayModel to string and writes it to the input.